server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/StorageManager.scala

```diff
@@ -441,50 +489,68 @@ private[worker] final class StorageManager(
           " working dirs. diskInfo $diskInfo")
         healthyWorkingDirs()
       }
-      if (dirs.isEmpty) {
+      if (dirs.isEmpty && hdfsFlusher.isEmpty) {
         throw new IOException(s"No available disks! suggested mountPoint $suggestedMountPoint")
       }
-      val index = getNextIndex()
-      val dir = dirs(index % dirs.size)
-      val shuffleDir = new File(dir, s"$appId/$shuffleId")
-      val file = new File(shuffleDir, fileName)
-      val mountPoint = DeviceInfo.getMountPoint(file.getAbsolutePath, mountPoints)
-
-      try {
-        shuffleDir.mkdirs()
-        val createFileSuccess = file.createNewFile()
-        if (!createFileSuccess) {
-          throw new RssException("create app shuffle data dir or file failed!" +
-            s"${file.getAbsolutePath}")
-        }
-        val shuffleKey = Utils.makeShuffleKey(appId, shuffleId)
-        val fileInfo = new FileInfo(file)
-        val fileWriter = new FileWriter(
+      val shuffleKey = Utils.makeShuffleKey(appId, shuffleId)
+      if (dirs.isEmpty) {
+        FileSystem.mkdirs(hdfsFs, new Path(s"$hdfsDir/$appId/$shuffleId"), hdfsPermission)
```

Comment:
What if the dir is already exists?

Reply:
Behavior is the same as a local file system.